packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -169,14 +227,17 @@ export class FauxClassGenerator {
     return decl.substr(decl.indexOf(' ') + 1);
   }
 
-  private generateFauxClasses(sobjects: SObject[], targetFolder: string) {
+  private generateFauxClasses(
+    sobjects: SObject[],
+    targetFolder: string
+  ): string {
     if (!this.createIfNeededOutputFolder(targetFolder)) {
-      console.log('no sobjects output folder ' + targetFolder);
-      return;
+      return nls.localize('no_sobject_output_folder_text', targetFolder);
     }
     for (const sobject of sobjects) {
       this.generateFauxClass(targetFolder, sobject);
     }
+    return '';
```

Comment:
Why are you returning an empty string to signify success? This doesn't seem like a good pattern.

Either move the check earlier `if (!this.createIfNeededOutputFolder(targetFolder)) {` or think of something else.

Reply:
agreed - not a good pattern - I was trying to avoid duplicating code, but I see a better way to do it.